spec/classes/init_spec.rb

```diff
@@ -1,98 +1,70 @@
+    let(:facts) { os_facts }
 
-    context 'when on Debian with the validate_sshd_file setting' do
-      let :facts do
-        {
-          :os => {
-            'family' => 'Debian'
-          },
-          'networking' => {
-            'interfaces' => {
-              'eth0' => {
-                'ip' => '10.0.0.1'
-              },
-              'eth1' => {
-                'ip' => '10.0.1.1'
+    context "on #{os}" do
+      context 'Server with a seperate sftp_server_init instance on Port 8022' do
+        let :params do
+          {
+            'server_instances' => {
+              'sftp_server_init' => {
+                'ensure' => 'present',
+                'options' => {
+                  'sshd_config' => {
+                    'Port' => 8022,
+                    'Protocol' => 2,
+                    'AddressFamily' => 'any',
+                    'HostKey' => '/etc/ssh/ssh_host_rsa_key',
+                    'SyslogFacility' => 'AUTH',
+                    'LogLevel' => 'INFO',
+                    'PermitRootLogin' => 'no',
+                  },
+                  'sshd_service_options' => '',
+                  'match_blocks' => {},
+                },
               },
-            }
+            },
           }
-        }
+        end
+
+        it { is_expected.to compile.with_all_deps }
+        it { is_expected.to contain_concat('/etc/ssh/sshd_config_sftp_server_init') }
+        it { is_expected.to contain_concat__fragment('sshd instance sftp_server_init config') }
+        it { is_expected.to contain_systemd__unit_file('sftp_server_init.service') }
+        it { is_expected.to contain_service('sftp_server_init.service') }
+        it { is_expected.to contain_ssh__server__instances('sftp_server_init') }
+        it { is_expected.to contain_class('ssh::client') }
+        it { is_expected.to contain_class('ssh::server') }
+        it { is_expected.to contain_concat('/etc/ssh/sshd_config').with_validate_cmd(nil) }
+        it { is_expected.to contain_resources('sshkey').with_purge(true) }
       end
-      let :params do
-        {
-          validate_sshd_file: true
-        }
+      context 'when on Debian with the validate_sshd_file setting' do
+        let :params do
+          {
+            validate_sshd_file: true
+          }
+        end
+
+        it { is_expected.to contain_class('ssh::client') }
+        it { is_expected.to contain_concat('/etc/ssh/sshd_config').with_validate_cmd('/usr/sbin/sshd -tf %') }
       end
+      context 'When on Debian without resource purging' do
+        let :params do
+          {
+            purge_unmanaged_sshkeys: false
+          }
+        end
 
-      it do
-        is_expected.to contain_class('ssh::client')
+        it { is_expected.not_to contain_resources('sshkey') }
       end
-      it do
-        is_expected.to contain_concat('/etc/ssh/sshd_config').with_validate_cmd('/usr/sbin/sshd -tf %')
+      context 'when on Debian with no other parameters' do
```

Comment:
this isn't Debian specific anymore:
```suggestion
      context 'with no other parameters' do
```